microbiomedata/submission-schema
# Use range union on env triad slots

Related to https://github.com/microbiomedata/issues/issues/718. I'll keep the issue open until these changes make it into a released version that is picked up by the submission portal.

### Summary

* Upgrade to `sheets-and-friends@0.5.3`
* Use new features in `sheets-and-friends` to replace the `range` of `env_broad_scale`, `env_local_scale`, and `env_medium`with an `any_of` construct that allows either the original enum _or_ a string. DataHarmonizer understands that it should present the enum permissible values in a dropdown menu. But at validation time it accepts either an enum permissible value or a string, which is what the schema says to do.
* Upgrade the version of `data-harmonizer` used by the interactive testing interface

### Testing

* Build the project artifacts with `make all`
* Start the interactive testing interface with `make dh-dev`
* Visit the testing interface at `http://localhost:5173` and from the toolbar select the `SoilInterface` template
* In the "broad-scale environmental context" column select an item from the dropdown menu (e.g. "arid biome [ENVO:01001838]"). Click the "Validate" button and verify that the cell is _not_ highlighted in red.
* Updated the cell to a valid that is _not_ part of the dropdown but does otherwise follow the pattern (e.g. "whatever biome [ENVO:123456]"). Click the "Validate" button and verify that the cell is _not_ highlighted in red.
* Update the cell to a string that does not follow the pattern (e.g. "whatever"). Click the "Validate" button and verify that the cell _is_ highlighted in red.
